### PR TITLE
[Data] e2e multi-node train benchmark

### PR DESCRIPTION
**Pull request by scottjlee**
## Why are these changes needed?

- Add resnet50 model to train loop (can be skipped with `--skip-train-model` arg). This is only enabled for image data, as the parquet format does not currently contain a mapping of image data to class labels.
- Add utility for mapping wnids to class labels, generate class IDs, etc.
- Add synthetic dataset (`--synthetic-data` flag), which is a randomly generated batch which is repeated 1000 times per iteration.
- Add checkpointing.

The accuracy is still quite low, in the single digit %. We will need to address this in a followup PR. For comparison, the mlcommons implementation runs to 75.9% accuracy after 41 epochs ([reference](https://github.com/mlcommons/training/tree/master/image_classification#quality-target)).

The throughput per worker is about 125 img/s, for both Ray Data and the synthetic data loader.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests - https://buildkite.com/ray-project/release/builds/1815
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/multi_node_train_benchmark.py [hunk 1]
```diff
@@ -169,6 +184,12 @@ def train_loop_per_worker():
         it = train.get_dataset_shard("train")
     device = train.torch.get_device()
 
+    # Setup the model
+    raw_model = resnet50(weights=ResNet50_Weights.DEFAULT)
```
Comment (stephanie-wang) [inline release/nightly_tests/dataset/multi_node_train_benchmark.py:188]: Won't this use the pretrained weights?

https://pytorch.org/vision/main/models/generated/torchvision.models.resnet50.html#torchvision.models.ResNet50_Weights

File: release/nightly_tests/dataset/multi_node_train_benchmark.py [hunk 2]
```diff
@@ -205,32 +238,106 @@ def train_loop_per_worker():
...
+        start_t = time.time()
+        num_batches = 0.0
+        total_loss = 0.0
+        for batch_idx, batch in enumerate(batch_iter):
             if not (args.use_torch or args.use_mosaic):
-                batch = batch["image"]
-            # `batch` should have tensor in `torch.Tensor` format.
-            num_rows += batch.size(dim=0)
-            if worker_rank == 0 and num_rows >= print_at:
+                num_rows += batch["image"].size(dim=0)
```
Comment (stephanie-wang) [inline release/nightly_tests/dataset/multi_node_train_benchmark.py:275]: Nit: might be nice to separate out some of these deep code blocks into helper methods but it's also fine if not.

**Review by stephanie-wang — APPROVED** [review]
Nice!
